### PR TITLE
Zoom to mouse cursor

### PR DESCRIPTION
**Pull request by bagnell**
Changes the default zoom behavior to zoom to the point under the mouse cursor. #302, #1060

### CODE REVIEW

**Comment by pjcozzi** [comment]
In 2D, zoom in close from a global view.  It feels too fast once we're close.

**Comment by pjcozzi** [comment]
Does this disable inertia for zoom in 2D and Columbus view?

**Comment by pjcozzi** [comment]
It seems like zoom inertia shuts off in 3D in some cases too.

**Comment by pjcozzi** [comment]
From a default view in 3D, a quick - but reasonable zoom - leads to too much jerk in the zoom path.

**Comment by bagnell** [comment]
>Does this disable inertia for zoom in 2D and Columbus view? It seems like zoom inertia shuts off in 3D in some cases too.

Do you have steps to reproduce this? Does it happen after switching between scene modes? I haven't seen it.

**Comment by pjcozzi** [comment]
Seems to happen to all seen modes after doing some middle click drag to 360 rotate around a point of interest.

**Comment by pjcozzi** [comment]
It's possible to get into a case like this:

http://localhost:1234/Apps/CesiumViewer/index.html?view=-160.70928356828844%2C-85.24792356385773%2C25857.484674075848%2C262.87914692976483%2C-7.844448971492795%2C329.66711361192637

image

From the home view, I left dragged to a point over the north pole so when I zoomed in the up vector was pointing south, then I went to the south pole.

File: Specs/Scene/ScreenSpaceCameraControllerSpec.js [hunk 1]
```diff
@@ -161,6 +161,11 @@ defineSuite([
         var ellipsoid = Ellipsoid.WGS84;
```
Comment (pjcozzi) [inline Specs/Scene/ScreenSpaceCameraControllerSpec.js]: Is this enough tests to cover all the new code?

**Comment by bagnell** [comment]
@pjcozzi This is ready for another look.

**Comment by pjcozzi** [comment]
I got this exception when I took this horizon view in beautiful North Wildwood:

image

**Comment by pjcozzi** [comment]
This doesn't reproduce it, but here is the link: http://localhost:1234/Apps/CesiumViewer/index.html?view=-74.79584484288888%2C38.994487937529996%2C22.77024267599088%2C50.52363135127272%2C0.2574589130355182%2C0.14517617284116282

**Comment by pjcozzi** [comment]
Crash when zooming into the baseball field in the middle of the screen:

image

http://localhost:1234/Apps/CesiumViewer/index.html?view=NaN%2CNaN%2CNaN%2CNaN%2CNaN%2CNaN

Note the NANs above.

The was was similar to:

http://localhost:1234/Apps/CesiumViewer/?view=-74.79695797656889%2C38.994452122368585%2C269.00627834334716%2C354.96315426043133%2C-33.760768723573946%2C359.98009431254536

**Comment by pjcozzi** [comment]
It takes two or three times to reproduce it, but start at the southeast corner of Wildwood:

http://localhost:1234/Apps/CesiumViewer/?view=-74.8662354080611%2C38.93857671653036%2C153.1966985275658%2C34.14668443896696%2C-17.946594252544585%2C0.11102167973795328

Then right click zoom the northeast tip of the North Wildwood beach and zoom in quickly.

Sometimes it is perfect; sometimes it slides along the ground awkwardly; and one time, it left me around South Carolina.

**Comment by pjcozzi** [comment]
Zooming over the north pole makes it easy to get into a situation where middle mouse drag is reversed or the horizon is tilted:

image

**Comment by pjcozzi** [comment]
From the home view in CV, click on Alaska then zoom in:

image

**Comment by pjcozzi** [comment]
Terrain was on in the CV scenario above.

**Comment by pjcozzi** [comment]
The CV zoom crash is pretty easy to reproduce on any zoom and without terrain.

**Comment by pjcozzi** [comment]
This crash is probably duplicate with the other `cartesian is required`:

image

**Comment by pjcozzi** [comment]
Can we make this work with 3D models using `pickPosition`?

**Comment by bagnell** [comment]
>Can we make this work with 3D models using pickPosition?

It already does, but the camera height has to be less than `ScreenSpaceCameraController.minimumPickingTerrainHeight`.

**Comment by bagnell** [comment]
@pjcozzi This is ready for another look. I couldn't reproduce many of the crashes after I fixed a couple. I'll try for a little while longer and post if I can.

**Comment by pjcozzi** [comment]
I can still reproduce this exception.  I was zoomed in to Yosemite with terrain on, then I left click panned a bit to a distant mountain peak, followed by right click zoom to the same peak:

image

**Comment by pjcozzi** [comment]
The final position it left me at was: http://localhost:8080/Apps/CesiumViewer/index.html?view=-93.05689156444894%2C42.091650485245914%2C5554.223648408619%2C86.81132417592867%2C-1.8757360391889446%2C0.19090024647699327

**Comment by pjcozzi** [comment]
Everything else looks good.  This is the only issue left that I can reproduce.

Also, merge in master.

**Comment by bagnell** [comment]
@pjcozzi This is ready for another look. The last crash is fixed. To reliably reproduce the crash, make sure terrain is on, the sky is at the center of the screen and right click zoom on terrain.

**Comment by pjcozzi** [comment]
Looks good!

**Pull request merged**